pinot-broker/src/main/java/org/apache/pinot/broker/routing/adaptiveserverselector/AdaptiveServerSelectorFactory.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.broker.routing.adaptiveserverselector;
+
+import com.google.common.base.Preconditions;
+import org.apache.pinot.core.transport.server.routing.stats.ServerRoutingStatsManager;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.apache.pinot.spi.utils.CommonConstants.Broker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class AdaptiveServerSelectorFactory {
+  private static final Logger LOGGER = LoggerFactory.getLogger(AdaptiveServerSelectorFactory.class);
+
+  private AdaptiveServerSelectorFactory() {
+  }
+
+  public static AdaptiveServerSelector getAdaptiveServerSelector(ServerRoutingStatsManager serverRoutingStatsManager,
+      PinotConfiguration pinotConfig) {
+    boolean enableStatsCollection =
+        pinotConfig.getProperty(Broker.AdaptiveServerSelector.CONFIG_OF_ENABLE_STATS_COLLECTION,
+            Broker.AdaptiveServerSelector.DEFAULT_ENABLE_STATS_COLLECTION);
+
```

Comment:
Add a log here to indicate whether stats collection is enabled / disabled ?

Reply:
We already have a log message to indicate this in ServerRoutingStatsManager.java